c2rust-analyze/src/panic_detail.rs

```diff
@@ -0,0 +1,170 @@
+use backtrace::Backtrace;
+use log::warn;
+use rustc_span::{Span, DUMMY_SP};
+use std::any::Any;
+use std::cell::Cell;
+use std::fmt::Write as _;
+use std::panic::PanicInfo;
+
+/// Detailed information about a panic.
+#[derive(Clone, Debug)]
+pub struct PanicDetail {
+    msg: String,
+    loc: Option<String>,
+    relevant_loc: Option<String>,
+    backtrace: Option<Backtrace>,
+    sp: Span,
+}
+
+impl PanicDetail {
+    /// Create a new `PanicDetail` containing only a message, with no location or backtrace
+    /// information.
+    pub fn new(msg: String) -> PanicDetail {
+        PanicDetail {
+            msg,
+            loc: None,
+            relevant_loc: None,
+            backtrace: None,
+            sp: DUMMY_SP,
```

Comment:
```suggestion
            sp: Default::default(),
```

Reply:
This is again less clear for the reader.  Rustc also mainly uses `DUMMY_SP` instead of `default()` for spans.